### PR TITLE
docs: Add ERC-4337 terms to the Glossary

### PR DESCRIPTION
**Pull request by germartinez**
## Context
This PR:
- Adds ERC-4337 terms to the Glossary

### CODE REVIEW

File: pages/glossary.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ See also:
 - [Account Abstraction](https://ethereum.org/en/roadmap/account-abstraction) on ethereum.org
 - [ERC-4337: Account Abstraction](https://www.erc4337.io) on erc4337.io
 
+## Bundler
+
+Bundlers are nodes that participate in the [ERC-4337](#erc-4337) standard who bundle [UserOperations](#user-operation) from an alternative mempool and submit them to the blockchain, paying for the associated transaction fees in advance, which are later refunded by the user account who proposed the user operation or by a [Paymaster](#paymaster).
+
+See also:
+- [Bundling](https://eips.ethereum.org/EIPS/eip-4337#bundling) process on ethereum.org
```
Comment (germartinez) [inline pages/glossary.md:25]:
```suggestion
- [Bundling process](https://eips.ethereum.org/EIPS/eip-4337#bundling on ethereum.org
```

File: pages/glossary.md [hunk 2]
```diff
@@ -61,6 +77,14 @@ A Safe owner is one of the accounts that control a given Safe. Only owners can m
 See also:
 - [OwnerManager.sol](https://github.com/safe-global/safe-smart-account/blob/main/contracts/base/OwnerManager.sol) on github.com
 
+## Paymaster
+
+Paymasters are smart contracts that allow an entity to pay for the gas fees of other users. This feature abstracts away the concept of gas fees by subsidizing them for users, allowing them to pay with ERC-20 tokens, and many other use cases.
```
Comment (tanay1337) [inline pages/glossary.md:82]: ", and many other use cases." -> Can we split this into another statement and add an example or two?

Comment (tanay1337) [inline pages/glossary.md:82]: Can we specify what entity means?

File: pages/glossary.md [hunk 3]
```diff
@@ -147,6 +171,14 @@ The threshold of a Safe account is a crucial configuration element that enables
 See also:
 - [Get the threshold](./sdk/protocol-kit/reference.md#getthreshold) and [change the threshold](./sdk/protocol-kit/reference.md#createchangethresholdtx) of a Safe with the Safe{Core} SDK on docs.safe.global
 
+## User Operation
+
+UserOperations are pseudo-transaction objects introduced by the [ERC-4337](#erc-4337) that users send to the UserOperation mempool. They wrap the users' transactions, and are sent to the [EntryPoint](#entrypoint) by [Bundlers](#bundler).
```
Comment (tanay1337) [inline pages/glossary.md:176]:
```suggestion
UserOperations are pseudo-transaction objects introduced by the [ERC-4337](#erc-4337) that users send to the UserOperation mempool. They wrap the users' transactions, and are sent to the [EntryPoint](#entrypoint) contract by [Bundlers](#bundler).
```

File: pages/glossary.md [hunk 4]
```diff
@@ -147,6 +171,14 @@ The threshold of a Safe account is a crucial configuration element that enables
 See also:
 - [Get the threshold](./sdk/protocol-kit/reference.md#getthreshold) and [change the threshold](./sdk/protocol-kit/reference.md#createchangethresholdtx) of a Safe with the Safe{Core} SDK on docs.safe.global
 
+## User Operation
```
Comment (tanay1337) [inline pages/glossary.md:174]: Should this be UserOperation instead?

Comment (germartinez) [reply pages/glossary.md:174]: This is something we still need to clarify. Multiple sources are using it differently
- User Operations
- UserOperations
- user operations

File: pages/glossary.md (hunk 2 above)
Comment (germartinez) [reply pages/glossary.md:82]: Entity is a company, project, person, or anybody interacting

Comment (germartinez) [reply pages/glossary.md:82]: Two examples were already shared:
- "subsidizing them for users".
- "allowing them to pay with ERC-20 tokens".

"and many other use cases" is used to not cover all the other cases in details, just emphasizing that more exist.

File: pages/glossary.md [hunk 5]
```diff
@@ -17,6 +17,14 @@ See also:
 - [Account Abstraction](https://ethereum.org/en/roadmap/account-abstraction) on ethereum.org
 - [ERC-4337: Account Abstraction](https://www.erc4337.io) on erc4337.io
 
+## Bundler
+
+Bundlers are nodes that participate in the [ERC-4337](#erc-4337) standard who bundle [UserOperations](#user-operation) from an alternative mempool and submit them to the blockchain, paying for the associated transaction fees in advance, which are later refunded by the user account who proposed the user operation or by a [Paymaster](#paymaster).
```
[the remark this answers is not in the document]
Comment (tanay1337) [reply pages/glossary.md:22]: Awesome!

File: pages/glossary.md (hunk 2 above)
Comment (tanay1337) [reply pages/glossary.md:82]: How about a small tweak?

`This feature abstracts away the concept of gas fees by subsidizing them for users, allowing them to pay with ERC-20 tokens, and enables many other use cases.`

Comment (tanay1337) [reply pages/glossary.md:82]: Thoughts on adding this in parathesis or potentially defining entity as a term?

File: pages/glossary.md (hunk 4 above)
Comment (tanay1337) [reply pages/glossary.md:174]: Should we ask in #protocol for opinions?

Comment (germartinez) [reply pages/glossary.md:174]: Well, these two can co-exist:
- "`UserOperation` objects"
- "user operations"

File: pages/glossary.md (hunk 2 above)
Comment (germartinez) [reply pages/glossary.md:82]: Maybe we can replace entity with "account"

Comment (tanay1337) [inline pages/glossary.md:82]:
```suggestion
Paymasters are smart contracts that allow an entity to pay for the gas fees of other users. This feature abstracts away the concept of gas fees by subsidizing them for users, allowing them to pay with ERC-20 tokens, and enables many other use cases.
```

**Review by tanay1337 — APPROVED** [review]

**Review by louis-md — APPROVED** [review]

**Pull request merged**